test/jdk/java/lang/ProcessBuilder/ArgCheck.java

```
@@ -0,0 +1,314 @@
+    private static final char NEWLINE = '\n';
+    private static final char BACKSLASH = '\\';
+
+    private static final String AMBIGUOUS_PROP_NAME = "jdk.lang.Process.allowAmbiguousCommands";
+    private static final String AMBIGUOUS_PROP_VALUE = System.getProperty(AMBIGUOUS_PROP_NAME);
+    private static final Boolean AMBIGUOUS_PROP_BOOLEAN = AMBIGUOUS_PROP_VALUE == null ? null :
+                                                          Boolean.valueOf(!"false".equals(AMBIGUOUS_PROP_VALUE));
+
+    private static final List<String> ECHO_JAVA_ARGS = Arrays.asList("java", "-classpath", TEST_CLASSES.toString(), "ArgCheck");
+    private static final List<String> ECHO_CMD_ARGS = Arrays.asList(ECHO_CMD_PATH);
+    private static final List<String> ECHO_VBS_ARGS = Arrays.asList("CScript", "/b", ECHO_VBS_PATH);
+
+    /**
+     * If zero arguments are supplied, run the test cases.
+     * If there are arguments, echo them to Stdout.
+     */
+    public static void main(String[] args) throws IOException {
+        if (args.length > 0) {
+            // Echo supplied arguments and exit
+            for (String arg : args)
+                System.out.println(arg);
+            return;
+        }
+
+        System.out.println("Java Version: " + System.getProperty("java.version"));
+
+        createFiles();
+
+        int errors = 0;
+        int success = 0;
+
+        for (CMD cmd : CASES) {
+            // If System property jdk.lang.process.allowAmbiguousCommands matches the case, test it
+            // If undefined, test them all
+            if (AMBIGUOUS_PROP_BOOLEAN == null ||
+                    AMBIGUOUS_PROP_BOOLEAN.booleanValue() == cmd.allowAmbiguous) {
+                try {
+                    testCommand(cmd);
+                    success++;
+                } catch (Exception ex) {
+                    ex.printStackTrace();
+                    errors++;
+                }
+            } else {
+                // skip unmatched cases
```

Can an "ignored" metric be added, along with "success" and "error" ?
I don't like silently skipping parts of a test.
Logging skipped test cases would work, too, but that might make the test output overly verbose.

---

Added a count of skipped tests and a final message.